### PR TITLE
Add another way to detect whether a viewer is expected.

### PR DESCRIPTION
**Pull request by cramforce**

### CODE REVIEW

File: src/service/viewer-impl.js [hunk 1]
```diff
@@ -223,7 +223,10 @@ export class Viewer {
         // origin is sometimes failed to be detected. Since failure mode
         // if we fail to initialize communication is very bad, we also check
         // for visibilityState.
-        && (this.params_['origin'] || this.params_['visibilityState'])
+        && (this.params_['origin']
+            || this.params_['visibilityState']
+            // Parent asked for viewer JS. We must be embedded.
+            || (this.win.location.search.indexOf('amp_js_v') != -1))
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: Do we have a ticket to remove these fallbacks?

Comment (cramforce) [inline src/service/viewer-impl.js]: This might turn out to be the right one :)

Comment (jridgewell) [inline src/service/viewer-impl.js]: Is this going to make it into our eventual public API? I thought it was just Google AMP Cache specific.

Comment (cramforce) [inline src/service/viewer-impl.js]: @chenshay's current work may, indeed, make this the public API. The param will trigger inserting an AMP extension.

**Pull request merged**